akka-http-caching/src/main/scala/akka/http/caching/Cache.scala

```diff
@@ -0,0 +1,63 @@
+package akka.http.caching
+
+import akka.annotation.ApiMayChange
+
+import scala.concurrent.{ Future, Promise }
+
+/**
+ * General interface implemented by all akka-http cache implementations.
+ */
+@ApiMayChange
+trait Cache[V] { cache ⇒
+
+  /**
+   * Returns either the cached Future for the given key or evaluates the given value generating
+   * function producing a `Future[V]`.
+   */
+  def apply(key: Any, genValue: () ⇒ Future[V]): Future[V]
+
+  /**
+   * Returns either the cached Future for the key or evaluates the given function which
+   * should lead to eventual completion of the promise.
+   */
+  def apply[U](key: Any, f: Promise[V] ⇒ U): Future[V] =
```

Comment:
`U` is unused here - would it make sense to make it specifically `Unit`? Or would that make it too hard to use?